src/index.ts

```diff
@@ -59,18 +59,27 @@ proxy.on('error', (err, req, res) => {
 
 const proxyServer = createServer((req, res) => {
   req.headers.host = stripPortHostHeader(req.headers.host);
+
   if (!req.headers.host) {
     res.writeHead(400, { 'Content-Type': 'text/plain' });
     res.write('Error: Request header host wasn\t specified');
     res.end();
     return;
   }
-  const proxyHost = proxyHosts.get(req.headers.host);
+
+  let proxyHost;
+  const urlParts = req.url ? req.url.split('/') : [''];
+
+  for (let i = urlParts.length - 1; i >= 0; i -= 1) {
+    proxyHost = proxyHosts.get(req.headers.host + urlParts.filter((_, j) => j <= i).join('/'));
+    if (proxyHost) break;
+  }
+
```

Comment:
This code section seems brittle. Please use the [URL API](https://developer.mozilla.org/en-US/docs/Web/API/URL_API) to get parts of urls.